cmd/kube-controller-manager/app/batch.go

```diff
@@ -41,14 +39,13 @@ func startJobController(ctx ControllerContext) (bool, error) {
 }
 
 func startCronJobController(ctx ControllerContext) (bool, error) {
-	if !ctx.AvailableResources[schema.GroupVersionResource{Group: "batch", Version: "v2alpha1", Resource: "cronjobs"}] {
+	if !ctx.AvailableResources[schema.GroupVersionResource{Group: "batch", Version: "v2alpha1", Resource: "cronjobs"}] &&
```

Comment:
Given client-go, it's basically impossible for a controller to tolerate two different API levels.  Are you sure this is doing what you expect?

Reply:
> Given client-go, it's basically impossible for a controller to tolerate two different API levels. Are you sure this is doing what you expect?

Yeah, if your controller is only going to work on the beta version (what it looks like), then you should only list the beta version as a gate.  Things don't go sideways until you switch to GA.

Reply:
Yup, I initially thought about working with both APIs, but failed. I forgot to update it.